File: backend/internal/model/group.go
package model

import (
	"github.com/gogf/gf/v2/os/gtime"
)

// Group 分组信息
type Group struct {
	Id          int64       `json:"id"          description:"分组ID"`
	Name        string      `json:"name"        description:"分组名称"`
	Description string      `json:"description" description:"分组描述"`
	CreatedAt   *gtime.Time `json:"createdAt"   description:"创建时间"`
	UpdatedAt   *gtime.Time `json:"updatedAt"   description:"更新时间"`
}

// GroupListReq 分组列表请求参数
type GroupListReq struct {
	Page     int    `json:"page"     v:"required#请输入页码"`   // 页码
	PageSize int    `json:"pageSize" v:"required#请输入每页数量"` // 每页数量
	Keyword  string `json:"keyword"`                       // 搜索关键词
}

// GroupListRes 分组列表返回结果
type GroupListRes struct {
	List     []Group `json:"list"`     // 分组列表
	Total    int     `json:"total"`    // 总数
	Page     int     `json:"page"`     // 页码
	PageSize int     `json:"pageSize"` // 每页数量
}

// GroupCreateReq 创建分组请求参数
type GroupCreateReq struct {
	Name        string `json:"name"        v:"required#请输入分组名称"` // 分组名称
	Description string `json:"description"`                      // 分组描述
}

// GroupUpdateReq 更新分组请求参数
type GroupUpdateReq struct {
	Id          int64  `json:"id"          v:"required#请输入分组ID"` // 分组ID
	Name        string `json:"name"        v:"required#请输入分组名称"` // 分组名称
	Description string `json:"description"`                      // 分组描述
}

File: frontend/src/api/group.ts
import { http } from "@/utils/http";

export interface GroupItem {
  id: number;
  name: string;
  description: string;
  deviceCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface GroupListReq {
  page: number;
  pageSize: number;
  keyword?: string;
}

export interface GroupListRes {
  code: number;
  message: string;
  data: {
    list: GroupItem[];
    total: number;
    page: number;
    pageSize: number;
  };
}

export interface GroupCreateReq {
  name: string;
  description?: string;
}

export interface GroupUpdateReq {
  id: number;
  name: string;
  description?: string;
}

export interface GroupDeleteReq {
  id: number;
}

// 获取分组列表
export function getGroupList(params: GroupListReq) {
  return http.request<GroupListRes>("get", "/api/groups/list", { params });
}

// 创建分组
export function createGroup(data: GroupCreateReq) {
  return http.request("post", "/api/groups/create", { data });
}

// 更新分组
export function updateGroup(data: GroupUpdateReq) {
  return http.request("put", "/api/groups/update", { data });
}

// 删除分组
export function deleteGroup(data: GroupDeleteReq) {
  return http.request("delete", "/api/groups/delete", { data });
}
